File: src/entities/vehicle/vehicle.cpp
#include "vehicle.hpp"

Vehicle::Vehicle(){}
Vehicle::Vehicle(int id) : id(id) {}
Vehicle::Vehicle(int id, std::string license_plate, std::string brand, std::string model,
                 int id_user, int id_vehicle_type) :  id(id),
                                                      license_plate(license_plate),
                                                      brand(brand),
                                                      model(model),
                                                      id_user(id_user),
                                                      id_vehicle_type(id_vehicle_type)
                                                      {}

void Vehicle::setId(int id){
  this->id = id;
}

void Vehicle::setLicensePlate(std::string license_plate){
  this->license_plate = license_plate;
}

void Vehicle::setIdUser(int id_user){
    this->id_user = id_user;
}

int Vehicle::getId() const{
    return id;
}

std::string Vehicle::getLicensePlate() const{
    return license_plate;
}

std::string Vehicle::getBrand() const{
    return brand;
}

std::string Vehicle::getModel() const{
    return model;
}

int Vehicle::getIdUser() const{
    return id_user;
}

int Vehicle::getIdVehicleType() const{
    return id_vehicle_type;
}

bool operator== (const Vehicle &v1, const Vehicle &v2)
{
    return v1.id == v2.id;
}

std::ostream& operator<<(std::ostream& os, const Vehicle& v)
{
    os << "Vehicle - Id: " << v.id << ", License Plate: " << v.license_plate << ", Brand: " << v.brand
        << ", Model: " << v.model << ", Id User: " << v.id_user << ", Id Vehicle Type: " << v.id_vehicle_type  << std::endl;
    return os;
}

VehicleException::VehicleException(const std::string & message):_message(message) {}
const char * VehicleException::what() const throw() {
      return _message.c_str();
}

File: uPark_client/entities/vehicle.py
class Vehicle:
    def __init__(self, id = None, license_plate = None, brand = None, model = None, id_user = None, id_vehicle_type = None):
        self._id = id
        self._license_plate = license_plate
        self._brand = brand
        self._model = model
        self._id_user = id_user
        self._id_vehicle_type = id_vehicle_type


    def set_id(self, id):
        self._id = id

    # def set_license_plate(self, license_plate):
    #     self._license_plate = license_plate

    def set_brand(self, brand):
        self._brand = brand

    def set_model(self, model):
        self._model = model

    # def set_id_user(self, id_user):
    #     self._id_user = id_user

    # def set_id_vehicle_type(self, id_vehicle_type):
    #     self._id_vehicle_type = id_vehicle_type


    def get_id(self):
        return self._id

    def get_license_plate(self):
        return self._license_plate

    def get_brand(self):
        return self._brand

    def get_model(self):
        return self._model

    def get_id_user(self):
        return self._id_user

    def get_id_vehicle_type(self):
        return self._id_vehicle_type

    def __eq__(self, other):
        if isinstance(other, Vehicle):
            return (self._id == other._id) or (self._license_plate == other._license_plate)
        return NotImplemented

    def __str__(self):
        vehicle_info = (f"Vehicle Id: {self._id}, license plate: {self._license_plate}, brand: {self._brand}, model: {self._model}, "
        f"id user: {self._id_user}, id vehicle type: {self._id_vehicle_type}")
        return vehicle_info
